Realm/RLMRealm.mm

```diff
@@ -706,6 +706,53 @@ - (void)cancelWriteTransaction {
     }
 }
 
+#ifdef REALM_ASYNC_WRITES
+
+- (BOOL)inAsyncWriteTransaction {
+    return _realm->is_in_async_transaction();
+}
+
+- (AsyncHandle)beginAsyncWriteTransaction:(void(^)())block {
+    return _realm->async_begin_transaction(block);
+}
+
+- (AsyncHandle)commitAsyncWriteTransaction {
+    return _realm->async_commit_transaction();
+}
+
+- (AsyncHandle)commitAsyncWriteTransaction:(nullable void(^)())doneBlock {
+    if (doneBlock) {
+        return _realm->async_commit_transaction(doneBlock);
+    }
+    return _realm->async_commit_transaction();
```

Comment:
Do you even need to check if `doneBlock` is not nil? It looks like you can pass nil into the core function.

Reply:
`async_commit_transaction` expecting done block as `&& the_done_block` and passing nil value will not be same as passing nil

Reply:
In that case, these blocks should not be `nullable` since you've already overloaded the method to not take a block param.